myhoard/controller.py

```diff
@@ -265,6 +266,17 @@ def mark_backup_requested(
                     return
             self.state_manager.update_state(backup_request=new_request)
 
+    def mark_backup_preservation(self, stream_id: str, preserve_until: Optional[datetime.datetime]) -> None:
+        backup_to_preserve = self._get_backup_by_stream_id(stream_id)
+        if not backup_to_preserve:
+            raise Exception(f"Stream {stream_id} was not found in completed backups.")
+
+        # should acquire lock in case the controller tries to purge the backup
+        with self.lock:
+            self._build_backup_stream(backup_to_preserve).mark_preservation(preserve_until=preserve_until)
+            # force refresh, this way we guarantee preserve_until is updated and backup does not get deleted
+            self._refresh_backups_list(force_refresh=True)
```

Comment:
No network IO must be done while holding the lock and this is also blocking network call from asyncio context which blocks the API.

Reply:
aff I forgot to remove that after previous clarification. Thanks!